### PR TITLE
Added extra fields for cronjob

### PR DESCRIPTION
**Pull request by psav**

### CODE REVIEW

File: apis/cloud.redhat.com/v1alpha1/clowdapp_types.go [hunk 1]
```diff
@@ -74,6 +75,12 @@ type Job struct {
 
 	// Defines the restart policy for the CronJob, defaults to never
 	RestartPolicy v1.RestartPolicy `json:"restartPolicy,omitempty"`
+
+	// Defines the concurrency policy for the CronJob, defaults to Forbid
```
Comment (BlakeHolifield) [inline apis/cloud.redhat.com/v1alpha1/clowdapp_types.go:79]: Should we default to Forbid or Allow? I think the kube standard is Allow. 
https://v1-16.docs.kubernetes.io/docs/tasks/job/automated-tasks-with-cron-jobs/#concurrency-policy

File: controllers/cloud.redhat.com/makers/maker.go [hunk 2]
```diff
@@ -245,6 +245,18 @@ func initJob(app *crd.ClowdApp, env *crd.ClowdEnvironment, cj *batch.CronJob, nn
 	cj.Spec.JobTemplate.ObjectMeta.Labels = labels
 	cj.Spec.JobTemplate.Spec.Template.ObjectMeta.Labels = labels
 
+	if job.ConcurrencyPolicy == "" {
+		cj.Spec.ConcurrencyPolicy = batch.ForbidConcurrent
```
Comment (BlakeHolifield) [inline controllers/cloud.redhat.com/makers/maker.go:249]: Same default question as above

File: controllers/cloud.redhat.com/makers/maker.go [hunk 3]
```diff
@@ -245,6 +245,18 @@ func initJob(app *crd.ClowdApp, env *crd.ClowdEnvironment, cj *batch.CronJob, nn
...
+	} else {
+		cj.Spec.ConcurrencyPolicy = job.ConcurrencyPolicy
+	}
+
+	if job.StartingDeadlineSeconds == nil {
+		cj.Spec.StartingDeadlineSeconds = utils.Int64Ptr(0)
+	} else {
+		cj.Spec.StartingDeadlineSeconds = job.StartingDeadlineSeconds
```
Comment (BlakeHolifield) [inline controllers/cloud.redhat.com/makers/maker.go:257]: Should this be the default? If != nil, add it. Else leave it out of the spec and take the default? https://v1-16.docs.kubernetes.io/docs/tasks/job/automated-tasks-with-cron-jobs/#starting-deadline

**Review by BlakeHolifield — COMMENTED** [review]
I know this is WIP, but I had some questions about the defaults. Also since this is open, can we update the CronJob schedule comment? If `schedule` isn't included, it breaks the App.

**Comment by psav** [comment]
Thanks for the comments @BlakeHolifield I hadn't finished with this one - but you helped me with the crafty research :)

**Pull request merged**